test/unit/messaging/messaging.spec.ts

```diff
@@ -608,6 +610,40 @@ describe('Messaging', () => {
         });
     });
 
+    it('should be fulfilled with a BatchResponse given array-like (issue #566)', () => {
+      const messageIds = [
+        'projects/projec_id/messages/1',
+        'projects/projec_id/messages/2',
+        'projects/projec_id/messages/3',
+      ];
+      mockedRequests.push(mockBatchRequest(messageIds));
+      const message = {
+        token: 'a',
+        android: {
+          ttl: 3600,
+        },
+      };
+      const arrayLike = new CustomArray();
+      arrayLike.push(message);
+      arrayLike.push(message);
+      arrayLike.push(message);
+      // Explicitly patch the constructor so that down compiling to ES5 doesn't affect the test.
+      // See https://github.com/firebase/firebase-admin-node/issues/566#issuecomment-501974238
+      // for more context.
+      arrayLike.constructor = CustomArray;
+
+      return messaging.sendAll(arrayLike)
```

Comment:
How do you validate the request arrayLike was property processed and converted to array?

Reply:
There's no way to validate it. But at least on non-ES5 targets, this test case fails if the SDK doesn't handle array-like objects correctly (it would throw the same error reported in #566).

Reply:
I see. Thanks for clarifying.